stx-rosetta.Dockerfile

```diff
@@ -0,0 +1,152 @@
+### Build blockstack-core-sidecar API
+FROM node:13.14.0-buster as build
+
+ARG API_TAG=v0.29.2
+
+RUN apt-get -y update && apt-get -y install openjdk-11-jre-headless
+
+WORKDIR /app
+
+RUN git clone -b $API_TAG --depth 1 https://github.com/blockstack/stacks-blockchain-api.git .
+# COPY ./rosetta-constants.ts /app/src/api/
+RUN grep process.env.STACKS_NETWORK /app/src/api/rosetta-constants.ts
+RUN echo "GIT_TAG=$(git tag --points-at HEAD)" >> .env
+RUN npm install && npm run build && npm prune --production
+
+### Build stacks-node binary
+
+FROM rust:stretch as stacks-node-build
+
+ARG STACKS_TAG=v23.0.0.10-krypton
+
+RUN mkdir -p /src /stacks
+WORKDIR /src
+RUN git clone -b $STACKS_TAG --depth 1 https://github.com/blockstack/stacks-blockchain.git .
+RUN rustup target add x86_64-unknown-linux-gnu
+RUN cargo build --release --workspace=./ --target x86_64-unknown-linux-gnu
+RUN cp -R /src/target/x86_64-unknown-linux-gnu/release/. /stacks
+
+### Fetch stacks-node binary
+
+### Begin building base image
+FROM ubuntu:focal
+
+ARG STACKS_NETWORK=testnet
+
+SHELL ["/bin/bash", "-c"]
+
+### Install utils
+RUN apt-get update
+RUN apt-get install -y sudo curl pslist
+
+### Set noninteractive apt-get
+RUN echo 'debconf debconf/frontend select Noninteractive' | debconf-set-selections
+
+### Storage goes in /data; see https://www.rosetta-api.org/docs/standard_storage_location.html
+RUN mkdir -p /data
+
+### stacky user ###
+# see https://docs.docker.com/develop/develop-images/dockerfile_best-practices/#user
+RUN useradd -l -u 33333 -G sudo -md /data/stacky -s /bin/bash -p stacky stacky \
+    # passwordless sudo for users in the 'sudo' group
+    && sed -i.bkp -e 's/%sudo\s\+ALL=(ALL\(:ALL\)\?)\s\+ALL/%sudo ALL=NOPASSWD:ALL/g' /etc/sudoers
+ENV HOME=/data/stacky
+WORKDIR $HOME
+USER stacky
+RUN sudo chown -R stacky:stacky $HOME
+RUN mkdir /data/stacky/.bashrc.d
+
+### Node.js
+# ENV NODE_VERSION=14.13.1
+ENV NODE_VERSION=13.14.0
+RUN curl -fsSL https://raw.githubusercontent.com/nvm-sh/nvm/v0.35.3/install.sh | bash \
```

Comment:
Curious to why `nvm` is used to install nodejs? As opposed to the typical apt-get command e.g.
```
curl -sL https://deb.nodesource.com/setup_14.x | bash -
apt-get install -y nodejs
```

Reply:
Ah, I'm assuming it's because of this requirement from rosetta docs?
> any required files must be downloaded from the internet with explicit versioning. If it is not possible to fetch a specific version, the hash of downloaded files must be compared with a constant defined in the Dockerfile.

Reply:
I used [follower.Dockerfile](https://github.com/blockstack/stacks-blockchain-api/blob/master/follower.Dockerfile) as the template for this dockerfile, and that uses nvm.  I can switch it to install the debs directly.

Reply:
Using the debs directly seems like a better choice (although rosetta dockerfile requirements seem ambiguous about what "external" files must be version pinned").

We can also switch to nodejs v14.x now that it's LTS.